guidelines/project-best-practices.md

```diff
@@ -0,0 +1,197 @@
+* Review [OpenSSF secure developer guide](https://github.com/ossf/wg-best-practices-os-developers/blob/main/docs/Concise-Guide-for-Developing-More-Secure-Software.md) and OpenSSF overview presentation to TOC ([charts](https://wiki.hyperledger.org/download/attachments/80775801/OpenSSF%20Hyperledger%2020230119.pdf?version=1&modificationDate=1674218724000&api=v2), [replay](https://youtu.be/0AFFIDKFNvc))
+* Review and obtain OpenSSF Best Practices Badge - see [criteria](https://bestpractices.coreinfrastructure.org/en/criteria/0)
+* Sign release artifacts (TBD) - see proposed [Security Artifact Signing task force](https://github.com/hyperledger/toc/issues/49)
+* Also see [2022 security task force](https://wiki.hyperledger.org/display/TF/Task+Force+Recommendations)
+
+## Documentation
+
+* Documentation should minimally target the following audiences
+  - User guide including Getting Started / Tutorial
+  - Project developer guide including coding guidelines, design docs, build instructions, test instructions
+  - Application developer guide
+* See the [documentation template lab](https://github.com/hyperledger-labs/documentation-template) for an example of what to include in documentation
+* [Documentation task force](https://wiki.hyperledger.org/display/TF/Documentation+task+force) will address Common styling guide, Recommended common publishing platform, Document best practices for creating documentation, etc.
+
+## Releases
+
+* Follow an established [Release taxonomy](https://toc.hyperledger.org/governing-documents/release-taxonomy.html) - either SemVer or CalVer; use consistent pre-release tags (e.g. -preview, -alpha, -beta)
+* Document release strategy and release process including required approvals
+* Document branch strategy, e.g. one branch per major.minor release works well so that it can be maintained in isolation while delivering major.minor.patch releases
+* Document Long-term support (LTS) release strategy - [Fabric example](https://github.com/hyperledger/fabric-rfcs/blob/main/text/0005-lts-release-strategy.md)
+* Use Github Actions to automate release process, e.g. publish artifacts and release notes upon drafting a GitHub release
+* Use [Reproducible builds](https://reproducible-builds.org/)
+* Sign release commits with [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (-S or --gpg-sign); sign release artifacts - see [Security Artifact Signing task force](https://github.com/hyperledger/toc/issues/49)
+* Release artifacts
+  - binaries attached to GitHub release
+  - docker images - transition from Dockerhub to GitHub Packages? See [data transfer and storage limits on GitHub Packages](https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages)
```

Comment:
@ryjones This one is still a question... I'm not sure if storage costs will become an issue if projects host multiple release and pre-release images on GitHub Packages. WDYT?

Reply:
For releases, we get 10 gigs a month of transfer. Over the last 30 days, we have transferred 1210 gigs from JFrog. I don't know how much of that is due to Fabric CI, though. I guess for now, let's leave DockerHub alone.

Reply:
ok